packages/side-nav/src/vaadin-side-nav-item.js

```diff
@@ -266,10 +271,9 @@ class SideNavItem extends SideNavChildrenMixin(DisabledMixin(ElementMixin(Themab
     if (this.path == null) {
       return false;
     }
-    return (
-      matchPaths(document.location.pathname, this.path) ||
-      this.pathAliases.some((alias) => matchPaths(document.location.pathname, alias))
-    );
+
+    const browserPath = `${document.location.pathname}${document.location.search}`;
```

Comment:
I couldn't find a way to stub `document.location` properties, as they are unwritable. For this reason, I didn't add unit tests for this change and only tested it manually.